open_instruct/reward_modeling.py

```diff
@@ -60,29 +57,45 @@
 
 @dataclass
 class Args:
-    # required dataset args
-    dataset_mixer: str = None
-    """A dictionary of datasets (local or HF) to sample from."""
-    dataset_train_splits: List[str] = None
+    # Dataset
+    dataset_mixer_list: List[str] = field(
+        default_factory=lambda: ["allenai/tulu-3-wildchat-reused-on-policy-8b", "1.0"]
+    )
+    """A list of datasets (local or HF) to sample from."""
+    dataset_mixer_eval_list: List[str] = field(default_factory=lambda: [])
+    """A list of datasets (local or HF) to sample from for evaluation."""
+    dataset_mixer_list_splits: List[str] = field(default_factory=lambda: ["train"])
     """The dataset splits to use for training"""
-    dataset_eval_mixer: Optional[str] = None
-    """A dictionary of datasets (local or HF) to sample from for evaluation"""
-    dataset_eval_splits: Optional[List[str]] = None
+    dataset_mixer_eval_list_splits: List[str] = field(default_factory=lambda: [])
     """The dataset splits to use for evaluation"""
-    dataset_mixer_dict: Optional[dict] = None
-    """The dataset mixer as a dictionary"""
-    dataset_eval_mixer_dict: Optional[dict] = None
-    """The dataset eval mixer as a dictionary"""
-
-    # common args
+    dataset_transform_fn: list[str] = field(default_factory=lambda: ["preference_tokenize_v1", "preference_filter_v1"])
```

Comment:
do we need to update reward modelling docs for the new args?

Reply:
I think it's nice for consistency, mostly because DPO / RM should share the same dataset preprocessing stuff.

Reply:
yeah 100% agree, just should update https://github.com/allenai/open-instruct/blob/main/docs/algorithms/reward_modeling.md

Reply:
Updated. Thanks for the catch!